build.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/sh
+
+fail() {
+    if [ "$NODE_DTRACE_PROVIDER_REQUIRE" == "" ]; then
+        echo "-----------------------------------------------------------------"
+        echo "Building dtrace-provider failed with exit code $1, falling back"
+        echo "on stub implementation."
+        echo ""
+        echo "Re-run install with V set in your environment to see the build"
+        echo "output, or NODE_DTRACE_PROVIDER_REQUIRE=hard to force an"
+        echo "installation failure."
+        echo "-----------------------------------------------------------------"
+    fi
+
+    if [ "$NODE_DTRACE_PROVIDER_REQUIRE" == "hard" ]; then
+        exit 1
+    else
+        exit 0
+    fi
+}
+
+buildUSDT() {
+    if [ "$NODE_DTRACE_PROVIDER_REQUIRE" == "" ] && [ "$V" == "" ]; then
+        exec 1> /dev/null
+        exec 2> /dev/null
+    fi
+
+    # GYP's MAKEFLAGS confuses libusdt's Makefile
+    unset MAKEFLAGS
+
+    # Ask node what architecture it's been built for ("target_arch" in
+    # config.gypi), and build libusdt to match.
+    #
+    # We use node from the path; npm will have adjusted PATH for us if
+    # necessary, otherwise we assume the user did so when building by
+    # hand.
+    #
+    # (This will need to change at the point that GYP is able to build
+    # node extensions universal on the Mac - for now we'll go with x86_64
+    # on a 64 bit Mac, because that's the default architecture in that
+    # situation.)
+    ARCH=`node -e "console.log(process.arch === 'x64' ? 'x86_64' : 'i386')"`
```

Comment:
Before this change, the module would catch type errors, apparently (from the code comment) to support node at versions 0.6.x? Is that not needed here?

Reply:
node 0.6.x and onwards has `process.arch`, and node 0.8.x onwards exposes the build configuration at `process.config`. The `process.arch` value is selected from the build configuration's `target_arch` though, so there was never any need to use both. I decided to use just `process.arch` which is supported in all of them.

node-dtrace-provider stopped supporting v0.6 and v0.8 in dtrace-provider@0.6.0, though, so it doesn't really matter much anymore.

Reply:
Sounds good to me!